### PR TITLE
Feature/lcb support dev

### PR DESCRIPTION
**Pull request by FrancoisLoubser-1**

### CODE REVIEW

File: layers/meta-balena-topic/recipes-support/bmap-tools/bmap-tools_3.6.bbappend [hunk 1]
```diff
@@ -0,0 +1 @@
+SRC_URI = "git://github.com/intel/${BPN};branch=main;protocol=https"
```
Comment (floion) [inline layers/meta-balena-topic/recipes-support/bmap-tools/bmap-tools_3.6.bbappend:1]: Do you use bmap-tools? What is the purpose?

File: layers/meta-balena-topic/recipes-core/images/balena-image.bbappend [hunk 2]
```diff
@@ -33,3 +33,9 @@ ROOTFS_POSTPROCESS_COMMAND += "topicbalena_rootfs_postprocess; "
 # We need to increase the rootfs size to make space for additional drivers in ROOTFS. 
 # BalenaOS image has a 700MB max size defined in 'image_types_balena.bbclass'
 IMAGE_ROOTFS_SIZE="409600"
+
```
Comment (floion) [inline layers/meta-balena-topic/recipes-core/images/balena-image.bbappend:37]: What is this needed for? Isn't `/etc/os-release` enough to provide with the info you need?

File: layers/meta-balena-topic/recipes-kernel/linux/linux-xlnx/0001-ASoc-adi-Kconfig-Remove-depends-on-for-ADI-reference.patch [hunk 3]
```diff
@@ -0,0 +1,28 @@
+From 91c0364479b932f473f72671c140d566dc758891 Mon Sep 17 00:00:00 2001
```
Comment (floion) [inline layers/meta-balena-topic/recipes-kernel/linux/linux-xlnx/0001-ASoc-adi-Kconfig-Remove-depends-on-for-ADI-reference.patch:1]: Shouldn't all these patches be in the BSP layer instead, the meta-topic submodule?

File: layers/meta-balena-topic/recipes-support/bmap-tools/bmap-tools_3.6.bbappend (hunk 1 above)
Comment (FrancoisLoubser-1) [reply layers/meta-balena-topic/recipes-support/bmap-tools/bmap-tools_3.6.bbappend:1]: Not explicitly. We just use the balenaos-img output.
The build does, however, generate some ..rootfs.bmap outputs by default in image_types_balena.bbclass, which could be used with the bmaptool for flashing.

This commit was more for cleaning up the build log to remove warnings.

File: layers/meta-balena-topic/recipes-core/images/balena-image.bbappend (hunk 2 above)
Comment (FrancoisLoubser-1) [reply layers/meta-balena-topic/recipes-core/images/balena-image.bbappend:37]: os-revision has a 'nice' version of BalenaOS, but a commit hash is the absolute truth.
I suppose this only helps while developing on a branch, where the os-revision stays fixed. On the 'master'/'production' BalenaOS revision does get stepped quite regularly.
Doesn't hurt to have it here, but I can easily remove it as well.

File: layers/meta-balena-topic/recipes-kernel/linux/linux-xlnx/0001-ASoc-adi-Kconfig-Remove-depends-on-for-ADI-reference.patch (hunk 3 above)
Comment (FrancoisLoubser-1) [reply layers/meta-balena-topic/recipes-kernel/linux/linux-xlnx/0001-ASoc-adi-Kconfig-Remove-depends-on-for-ADI-reference.patch:1]: The machine (lcbzu9) that has been added consists of a Topic SoM module on top of the client's custom carrier board.
It is therefore technically not part of the Topic PCBs and not supported by meta-topic.

If you don't think these fit in meta-balena-topic, a workable suggestion might be to add something like "meta-lowpad".
Then I would move everything required to support the lowpad carrier board, including these patches, the machine.conf, fpga & device tree there.

Do you agree?

Comment (floion) [reply layers/meta-balena-topic/recipes-kernel/linux/linux-xlnx/0001-ASoc-adi-Kconfig-Remove-depends-on-for-ADI-reference.patch]: I would rather we move all these to a meta-lowpad layer.
But if you are in a hurry now, we can merge them like this and you can move all lcbzu9 code at a later time in the new layer.

**Review by floion — APPROVED** [review]

**Pull request merged**